### PR TITLE
Add script to extract locale strings

### PR DESCRIPTION
**Pull request by VeraZab**

### CODE REVIEW

**Comment by VeraZab** [comment]
@bpostlethwaite ready for a look
cc @alexcjohnson

File: utils/findLocaleStrings.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    if (!EXIT_CODE) {
+      const strings = Object.keys(dict)
+        .sort()
+        .map(k => k + spaces(maxLen - k.length) + '  // ' + dict[k])
+        .join('\n');
+      const pathToTranslationKeys = path.join(__dirname, 'translationKeys.txt');
+      fs.writeFile(pathToTranslationKeys, strings);
+      console.log('ok find_locale_strings');
```
Comment (bpostlethwaite) [inline utils/findLocaleStrings.js]: :zap:

Comment (bpostlethwaite) [reply utils/findLocaleStrings.js]: or maybe not? Dunno

File: utils/findLocaleStrings.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+              maxLen = Math.max(maxLen, strNodeValue.length);
+              hasTranslation = true;
+            }
+          }
+        },
+      });
+    });
+
+    if (!hasTranslation) {
+      console.error('Found no translations.');
```
Comment (bpostlethwaite) [inline utils/findLocaleStrings.js]: logError?

File: utils/findLocaleStrings.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+let EXIT_CODE = 0;
+const localizeRE = /(^|[\.])(_|localize)$/;
+
+findLocaleStrings();
+
+function findLocaleStrings() {
+  glob(srcGlob, (err, files) => {
+    if (err) {
+      EXIT_CODE = 1;
+      console.log(err);
```
Comment (bpostlethwaite) [inline utils/findLocaleStrings.js]: console.error ?

Comment (bpostlethwaite) [reply utils/findLocaleStrings.js]: or this can simply be `if (err) throw(err)`

Comment (bpostlethwaite) [reply utils/findLocaleStrings.js]: that will automatically use exitcode 1 and console.error anyway.

Comment (bpostlethwaite) [reply utils/findLocaleStrings.js]: Oh I see there is an error handler below that catches these things.

**Comment by VeraZab** [comment]
yeah, I can remove the translations, I believe @alexcjohnson committed them in the plotly.js repo, but it's true, it's not really necessary

File: utils/findLocaleStrings.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+                  path.node,
+                  `The localization function takes a string as argument, instead it received a ${typeof strNodeValue}`
+                );
+              }
+            }
+
+            if (!dict[strNodeValue]) {
+              dict[strNodeValue] =
+                filePartialPath + ':' + strNode.loc.start.line;
+              maxLen = Math.max(maxLen, strNodeValue.length);
```
Comment (alexcjohnson) [inline utils/findLocaleStrings.js]: might want to put a limit on this... 80? so a few of the longest strings would not have the line-number comment lined up with the rest (just give them at least 2 spaces), but the majority of them would and it'd be closer and easier to use.

File: utils/translationKeys.txt [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+Connect Gaps                                                                                                 // /DefaultEditor.js:162
+Continue                                                                                                     // /components/widgets/text_editors/MultiFormatTextEditor.js:197
+Continuing will convert your LaTeX expression into raw text.                                                 // /components/widgets/text_editors/MultiFormatTextEditor.js:121
+Continuing will convert your note to LaTeX-style text.                                                       // /components/widgets/text_editors/MultiFormatTextEditor.js:116
+Continuing will remove your expression.                                                                      // /components/widgets/text_editors/MultiFormatTextEditor.js:126
+Custom                                                                                                       // /DefaultEditor.js:179
+Custom colors                                                                                                // /components/widgets/ColorPicker.js:52
```
Comment (alexcjohnson) [inline utils/translationKeys.txt]: Looks like we've generally capitalized all words of titles, so `Custom Colors`? `Default Colors`? Less sure about `Enter Link URL`...

File: utils/translationKeys.txt [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+Show                                                                                                         // /DefaultEditor.js:233
+Size                                                                                                         // /DefaultEditor.js:131
+Size and Spacing                                                                                             // /DefaultEditor.js:137
+Symbol                                                                                                       // /DefaultEditor.js:132
+Text                                                                                                         // /DefaultEditor.js:364
+Text Attributes                                                                                              // /DefaultEditor.js:82
+The anchor point determines which side of the annotation's positioning coordinates refer to.                 // /DefaultEditor.js:256
+The positioning inputs are relative to the anchor points on the text box                                     // /DefaultEditor.js:389
```
Comment (alexcjohnson) [inline utils/translationKeys.txt]: `.`

Comment (bpostlethwaite) [reply utils/translationKeys.txt]: This is a nice way to correct all the text in a program.

File: utils/findLocaleStrings.js (hunk 4 above)
Comment (VeraZab) [reply utils/findLocaleStrings.js]: I kind of like that they're all aligned, even if there are some long stings, I find it's more consistent for my eyes :) and makes me notice the super long strings, i.e. may be a sign that they'd need to be shortened...

File: utils/translationKeys.txt (hunk 6 above)
Comment (VeraZab) [reply utils/translationKeys.txt]: yup, agreed, making corrections now :)

**Comment by VeraZab** [comment]
ok, made the changes @bpostlethwaite @alexcjohnson mergeable?

**Comment by alexcjohnson** [comment]
💃

**Comment by bpostlethwaite** [comment]
nice work! :dancer:

**Pull request merged**